core/lib/progressbar.ts

```diff
@@ -0,0 +1,191 @@
+
+export type ProgressbarContext = WidgetSlotContext<ProgressbarWidget>;
+
+export interface ProgressbarCommonPropsAndState {
+	/**
+	 * the minimum value
+	 * @defaultValue 0
+	 */
+	minimum: number;
+
+	/**
+	 * the maximum value
+	 * @defaultValue 100
+	 */
+	maximum: number;
+
+	/**
+	 * the current value
+	 * @defaultValue 0
+	 */
+	value: number;
+
+	/**
+	 * the aria label
+	 */
+	ariaLabel: string;
+
+	/**
+	 * class to add to the content
+	 */
+	className: string;
+
+	/**
+	 * global template for the Progressbar content
+	 */
+	slotContent: SlotContent<ProgressbarContext>;
+	/**
+	 * label of the progress
+	 */
+	slotDefault: SlotContent<ProgressbarContext>;
+	/**
+	 * height of the progressbar, can be any valid css height value
+	 */
+	height: string;
+	/**
+	 * if `true`, display the current percentage in the `xx%` format
+	 */
+	showPercentage: boolean;
+	/**
+	 * if `true`, shows a striped progressbar
+	 */
+	striped: boolean;
+	/**
+	 * if `true`, animates a striped progressbar
+	 */
+	animated: boolean;
+}
+
+export interface ProgressbarState extends ProgressbarCommonPropsAndState {
+	/**
+	 * percentage of completion
+	 */
+	percentage: number;
+	/**
+	 * `true` if the value is above its minimum value
+	 */
+	started: boolean;
+	/**
+	 * `true` if the value has reached its maximum value
+	 */
+	finished: boolean;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ProgressbarProps extends ProgressbarCommonPropsAndState {}
+
+export interface ProgressbarApi {
+	/**
+	 * Reset the progress bar to the minimum value
+	 */
+	reset(): void;
+
+	/**
+	 * Set the progress bar to the maximum value.
+	 */
+	end(): void;
+}
+
+export type ProgressbarWidget = Widget<ProgressbarProps, ProgressbarState, ProgressbarApi, object, object>;
+
+const defaultConfig: ProgressbarProps = {
+	minimum: 0,
+	maximum: 100,
+	value: 0,
+	ariaLabel: 'Progressbar',
+	className: '',
+	slotContent: undefined,
+	slotDefault: undefined,
+	height: '',
+	showPercentage: false,
+	striped: false,
+	animated: false,
+};
+
+/**
+ * Retrieve a shallow copy of the default Progressbar config
+ * @returns the default Progressbar config
+ */
+export function getProgressbarDefaultConfig(): ProgressbarProps {
+	return {...defaultConfig};
+}
+
+const configValidator: ConfigValidator<ProgressbarProps> = {
+	minimum: typeNumber,
+	maximum: typeNumber,
+	value: typeNumber,
+	ariaLabel: typeString,
+	className: typeString,
+	height: typeString,
+	showPercentage: typeBoolean,
+	striped: typeBoolean,
+	animated: typeBoolean,
+};
+
+/**
+ * Create an ProgressbarWidget with given config props
+ * @param config - an optional progress bar config
+ * @returns an ProgressbarWidget
+ */
+export function createProgressbar(config?: PropsConfig<ProgressbarProps>): ProgressbarWidget {
+	const [
+		{
+			minimum$,
+			maximum$: _dirtyMaximum$,
+			// dirty value that needs adjustment:
+			value$: _dirtyValue$,
+			ariaLabel$,
```

Comment:
No need to create local stores if there are just "proxied" to the return, just get the ones needed in the internal logic.